src/test/java/org/minions/devfund/angela/SquarelotronTest.java

```diff
@@ -0,0 +1,228 @@
+package org.minions.devfund.angela;
+
+import org.junit.Before;
+import org.junit.Test;
+
+import static org.junit.Assert.assertArrayEquals;
+import static org.junit.Assert.assertEquals;
+
+/**
+ * Test for {@link Squarelotron} .
+ */
+public class SquarelotronTest {
+
+    private Squarelotron squarelotron;
+
+    /**
+     * Enumerable for numbers.
+     */
+    public enum Numbers {
+        ONE(1),
+        TWO(2),
+        THREE(3),
+        FOUR(4),
+        FIVE(5),
+        SIX(6),
+        SEVEN(7),
+        EIGHT(8),
+        NINE(9),
+        TEN(10),
+        ELEVEN(11),
+        TWELVE(12),
+        THIRTEEN(13),
+        FOURTEEN(14),
+        FIFTEEN(15),
+        SIXTEEN(16);
+
+        private int number;
+
+        /**
+         * Constructor for enum.
+         *
+         * @param value int numeral.
+         */
+        Numbers(final int value) {
+            number = value;
+        }
+
+        /**
+         * {@inheritDoc}
+         */
+        @Override
+        public String toString() {
+            return String.valueOf(number);
+        }
+
+        /**
+         * Gets the number.
+         *
+         * @return int number value.
+         */
+        public int getNumber() {
+            return number;
+        }
+    }
+
+    /**
+     * Preconditions.
+     */
+    @Before
+    public void setup() {
+        int size = Numbers.FOUR.getNumber();
+        squarelotron = new Squarelotron(size);
+    }
+
+    /**
+     * Test if Upside Down Flip is working.
+     */
+    @Test
+    public void testUpsideDownFlip() {
+        final Squarelotron squareResult = squarelotron.upsideDownFlip(2);
+        assertEquals(Numbers.TEN.getNumber(), squareResult.getSquarematrix()[1][1]);
+        assertEquals(Numbers.ELEVEN.getNumber(), squareResult.getSquarematrix()[1][2]);
+        assertEquals(Numbers.SIX.getNumber(), squareResult.getSquarematrix()[2][1]);
+        assertEquals(Numbers.SEVEN.getNumber(), squareResult.getSquarematrix()[2][2]);
+    }
+
+    /**
+     * Test if Main Diagonal Flip is working.
+     */
+    @Test
+    public void testMainDiagonalFlip() {
+        final Squarelotron squareResult = squarelotron.mainDiagonalFlip(2);
+        assertEquals(Numbers.SIX.getNumber(), squareResult.getSquarematrix()[1][1]);
+        assertEquals(Numbers.TEN.getNumber(), squareResult.getSquarematrix()[1][2]);
+        assertEquals(Numbers.SEVEN.getNumber(), squareResult.getSquarematrix()[2][1]);
+        assertEquals(Numbers.ELEVEN.getNumber(), squareResult.getSquarematrix()[2][2]);
```

Comment:
Your implementation for upsideDownFlip and mainDiagonalFlip is not working when the ring is 2 or 3 and so on.
For example these scenarios are failing.
```java
    @Test
    public void testUpsideDownFlipDimension4Ring2() {
        final int ringNumber = 2;
        final int dimension = 4;
        final Squarelotron squarelotron = new Squarelotron(dimension);
        final int[][] expectedMatrix = new int[][]{{1, 2, 3, 4}, {5, 10, 11, 8}, {9, 6, 7, 12}, {13, 14, 15, 16}};
        Squarelotron result = squarelotron.upsideDownFlip(ringNumber);
        assertArrayEquals(expectedMatrix, result.getSquarematrix());
    }

    @Test
    public void testMainDiagonalFlipDimension4Ring2() {
        final int ringNumber = 2;
        final int dimension = 4;
        final Squarelotron squarelotron = new Squarelotron(dimension);
        final int[][] expectedMatrix = new int[][]{{1, 2, 3, 4}, {5, 6, 10, 8}, {9, 7, 11, 12}, {13, 14, 15, 16}};
        Squarelotron result = squarelotron.mainDiagonalFlip(ringNumber);
        assertArrayEquals(expectedMatrix, result.getSquarematrix());
    }
```
Also Consider writting your unit test like these examples. :)